src/main/webapp/app/shared/import/users-import-dialog.component.ts

```diff
@@ -0,0 +1,267 @@
+        private examManagementService: ExamManagementService,
+        private courseManagementService: CourseManagementService,
+        private translateService: TranslateService,
+    ) {}
+
+    ngOnDestroy(): void {
+        this.dialogErrorSource.unsubscribe();
+    }
+
+    private resetDialog() {
+        this.usersToImport = [];
+        this.notFoundUsers = [];
+        this.hasImported = false;
+    }
+
+    async onCSVFileSelect(event: any) {
+        if (event.target.files.length > 0) {
+            this.resetDialog();
+            this.usersToImport = await this.readUsersFromCSVFile(event, event.target.files[0]);
+        }
+    }
+
+    /**
+     * Reads users from a csv file into a list of StudentDTOs
+     * The column "registrationNumber" is mandatory since the import requires it as an identifier
+     * @param event File change event from the HTML input of type file
+     * @param csvFile File that contains one user per row and has at least the columns specified in csvColumns
+     */
+    private async readUsersFromCSVFile(event: any, csvFile: File): Promise<StudentDTO[]> {
+        let csvUsers: CsvUser[] = [];
+        try {
+            this.isParsing = true;
+            this.validationError = undefined;
+            csvUsers = await this.parseCSVFile(csvFile);
+        } catch (error) {
+            this.validationError = error.message;
+        } finally {
+            this.isParsing = false;
+        }
+        if (csvUsers.length > 0) {
+            this.performExtraValidations(csvFile, csvUsers);
+        }
+        if (this.validationError) {
+            event.target.value = ''; // remove selected file so user can fix the file and select it again
+            return [];
+        }
+        return csvUsers.map(
+            (users) =>
+                ({
+                    registrationNumber: users[csvColumns.registrationNumber] || users[csvColumns.matrikelNummer] || users[csvColumns.matriculationNumber] || '',
+                    login: users[csvColumns.login] || users[csvColumns.username] || users[csvColumns.user] || users[csvColumns.benutzer] || users[csvColumns.benutzerName] || '',
+                    firstName: users[csvColumns.firstName] || users[csvColumns.firstNameOfUser] || '',
+                    lastName: users[csvColumns.lastName] || users[csvColumns.familyName] || users[csvColumns.familyNameOfUser] || '',
+                } as StudentDTO),
+        );
+    }
+
+    /**
+     * Performs validations on the parsed users
+     * - checks if values for the required column {csvColumns.registrationNumber} are present
+     *
+     * @param csvFile File that contains one user per row and has at least the columns specified in csvColumns
+     * @param csvUsers Parsed list of users
+     */
+    performExtraValidations(csvFile: File, csvUsers: CsvUser[]) {
+        const invalidUserEntries = this.computeInvalidUserEntries(csvUsers);
+        if (invalidUserEntries) {
+            const maxLength = 30;
```

Comment:
Could we extract it to a constant and describe what `30` represents?

Reply:
This constant is only used to now overwhelm the user in case the CSV is having to many invalid entries, so i dont think we need to extract it into the class as a constant. But I can do so if you want to, let me know 👍